kyaml/fn/framework/example_test.go

```diff
@@ -221,7 +222,7 @@ func ExampleTemplateProcessor_generate_files() {
 		// Templates
 		ResourceTemplates: []framework.ResourceTemplate{{
 			Templates: framework.TemplatesFromFile(
-				filepath.Join("testdata", "example", "templatefiles", "deployment.template"),
+				pkger.Include("/fn/framework/testdata/example/templatefiles/deployment.template"),
```

Comment:
Is there a reason not to do something like `pkger.Include(filepath.Join("fn"....`, to make it OS-agnostic?

Reply:
In this specific test situation, not really, because it isn't _really_ using pkger. But pkger uses static analysis to figure out what to package: it's reading string literals. It's also why there's a leading slash: it treats the module root directory as the FS root.

Reply:
So this wouldn't work if replaced with `Join`?  Because pkger is looking for these paths?  Never used it, looks interesting.

If so please put a comment in there because we sometimes get driveby PR's that would make such a change in the name of os portabilty

Reply:
It would be fine to replace it with a join in this test code, because there's no `pkged.go`, which means it will fall through to reading the real FS. In cases like this where you're not using pkger, the only weird requirement is to express the path as an absolute path that uses the module root as the root of the FS. This is also true if you're using pkger but NOT its auto-inclusion feature, i.e. your pkger invocation specifies what to include manually.

The problem is when you're using pkger AND specifically using its auto-detection feature (`Dir` and `Include`) to determine what gets included in `pkged.go`. In that case, you can't use `Join`. The auto-detection only works if each path is literally expressed and _directly_ surrounded by either `pkger.Dir()` or `pkger.Include()`. I realized in explaining this that the following works fine though: `TemplatesFromFile(filepath.FromSlash(pkger.Include("/kyaml/fn/framework/foo/bar.template.yaml")))`, because `Include` just returns the string verbatim and remains directly around it in this formulation. 

I've added some explanations/examples to the function docs and changed the tests to not bother mentioning pkger.

Sidenote: I haven't looked into it yet, but I think I'd like to replace pkger with https://golang.org/pkg/embed/ in the framework. Pkger is neat and works nicely once you figure it out, but it also has its quirks.